tests/unit_tests/fn/test_extents.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <gridtools/fn/extents.hpp>
+
+#include <type_traits>
+
+#include <gtest/gtest.h>
+
+#include <gridtools/common/integral_constant.hpp>
+#include <gridtools/common/tuple_util.hpp>
+
+namespace gridtools::fn {
+    namespace {
+        template <class A,
+            class B,
+            class CommonValueType = std::common_type_t<typename A::value_type, typename B::value_type>>
+        struct integral_constant_equal
```

Comment:
this predicate is sfinae unfriendly and also it doesn't test that `A` and `B` are `is_gt_integral_constant`

Reply:
I guess that what you really need in this file is `is_gt_integral_constant_of`

Reply:
Actually I don't know why I thought this make sense.
I replaced the tests by `static_assert(some_integral_constant::value ==some_integral);`